params/params.go

```diff
@@ -24,11 +22,6 @@ func (u *Username) UnmarshalText(b []byte) error {
 	if utf8.RuneCount(b) > 256 {
 		return errgo.New("username longer than 256 characters")
 	}
-	for _, part := range bytes.Split(b, []byte("@")) {
-		if !names.IsValidUserName(string(part)) {
```

Comment:
I agree that we shouldn't use juju/names to validate this, but what are the security implications of not validating this string at all?

Does this introduce any kind of subtle vulnerability to homoglyph attacks, invisible spaces, etc?

I guess what I'm missing though is the context. If the client doesn't validate the username but it still gets validated by the candid server, maybe this is OK?

Reply:
I don't think this significantly increases the attack surface. The username is only used to find existing users in the database, which will require a byte for byte name match.